### ISSUE TITLE
Suspend not working in 21.04 kernel 5.13.0-7620 for System76 Galago Pro

### ISSUE THREAD

**Issue by curiousercreative** [pop-os/pop#2006]
**Distribution (run `cat /etc/os-release`):**
 ```
NAME="Pop!_OS"
VERSION="21.04"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 21.04"
VERSION_ID="21.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=hirsute
UBUNTU_CODENAME=hirsute
LOGO=distributor-logo-pop-os
```

**Issue/Bug Description:**
System76 Galago Pro fails to enter suspend

**Steps to reproduce (if you know):**
1. With latest 21.04 + 5.13.0-7620 booted, attempt to suspend from the power menu
AR/ER: screen goes black, networking halted
ER: green power LED goes from solid to blinking, power consumption drops to 2% per hour
AR: green power LED remains solid, power consumption 10-15% per /hour

**Other Notes:**
- Confirmed suspend works fine when booted into old kernel (5.13.0-7614) and recovery
- Per System76 support direction, NVIDIA drivers have already been reinstalled (support case is ongoing)
- Likely related to [this other report](https://askubuntu.com/questions/1375825/kernel-5-15-and-5-14-fails-to-suspend)

**Comment by jacobgkau** [comment]
I am not seeing this on our lab galp5 with 1650 Ti. I'm seeing the system suspend, continue having its power LED blink until I resume, and then successfully resume. this is with kernel 5.13.0-7620 and NVIDIA 470.82 on Pop!_OS 21.04.

>  AR/ER: screen goes black, networking halted
>  ER: green power LED goes from solid to blinking, power consumption drops to 2% per hour
>  AR: green power LED remains solid, power consumption 10-15% per /hour

Can you please explain what "AR" and "ER" mean in this report?

> Likely related to [this other report](https://askubuntu.com/questions/1375825/kernel-5-15-and-5-14-fails-to-suspend)

That user of an ASUS system with an i9 processor is only having problems with 5.14+, not 5.13, and they're unable to install the NVIDIA driver on 5.13. Can you please explain why you think it's related?

Does this affect all graphics modes? Does it affect a live disk or up-to-date Recovery partition of 21.04? Are you running customized firmware?

**Comment by curiousercreative** [comment]
@jacobgkau thanks the fast attention:
- AR = actual/observed result, ER = expected result
- lemme try booting a recovery or live installer with the same kernel. As mentioned, previous kernel and same system works as expected.
- I'm also configured with a 1650 Ti, latest drivers. I haven't tested various graphics modes, just integrated.

**Comment by jacobgkau** [comment]
Are you using a USB-C display or any other Thunderbolt or USB-C devices?

**Comment by curiousercreative** [comment]
> Are you using a USB-C display or any other Thunderbolt or USB-C devices?

Not while observing this behavior. I'm aware of the USB-C suspend firmware bug and this doesn't appear to be related (nor firmware).

**Comment by curiousercreative** [comment]
@jacobgkau per support advice, I fresh installed beta with kernel 5.15 and the issue persists there. Here's my thinking:
1. Not OS install related as it reproduces on 5.13.0-7620 but not 5.13.0-7614 with otherwise identical system install. Also reproduces on fresh OS install of 5.15
2. Not firmware related (for the above reasons).
3. Not model specific (since your identical galp5 doesn't repro)

All of this suggests a hardware specific kernel regression to me. What else is there to explore? The only hardware that's specific to my galp5 vs yours would be storage (ADATA SX8200NP 960GB) and memory (2x Crucial 16GB 3200 MHz).

**Comment by curiousercreative** [comment]
This [patch](https://lkml.kernel.org/lkml/20210712061005.392727896@linuxfoundation.org/) could be suspect? I don't know how to, but someone could try bisecting on that. Other ideas?

**Comment by curiousercreative** [comment]
If this issue isn't resolved sooner, I'll have a framework laptop in the next few weeks that I plan to install the NVMe and memory in, so that may further confirm or deny my hypothesis

**Comment by curiousercreative** [comment]
As a follow-up to my last comment, I've migrated my system (NVMe with OS install and memory) to the framework laptop and `5.13.0-7620-generic` suspends without an issue.

**Comment by jacobgkau** [comment]
> The only hardware that's specific to my galp5 vs yours would be storage (ADATA SX8200NP 960GB) and memory (2x Crucial 16GB 3200 MHz).

Does the issue occur with the storage and RAM that were originally shipped in your system? (I understand the aftermarket parts work in your Framework machine, I'm trying to ascertain whether this is a clear-cut hardware defect with the Galago or some kind of compatibility issue.)

**Comment by curiousercreative** [comment]
@jacobgkau unfortunately a few things prevent me from testing this:
1. I've sold the original WD Blue 500GB NVMe
2. My galp5 was just shipped in to the repair center for a warranty repair (chassis crack).

You might reach out to the CA repair center for further testing? I'm happy to provide the RMA # or anything else that might be helpful.

**Comment by jacobgkau** [comment]
Have you communicated these issues to the support team via your support ticket? I would recommend doing that, GitHub is not a place for support transactions to take place.

**Comment by jacobgkau** [comment]
I have located your support ticket and will ask our support team to have suspend/resume tested while this is in.

**Comment by jacobgkau** [comment]
I am closing this issue since you have a support case open and this issue is not occurring on a lab unit of the same model. This can be re-opened if further investigation shows that this is somehow a Pop!_OS-related issue.

**Comment by curiousercreative** [comment]
@jacobgkau Understood. I think I opened a support ticket before this issue, but I can no longer really track my support tickets with the new ticketing system.

**Comment by curiousercreative** [comment]
@leviport @jacobgkau after receiving my galp5 back today, I've isolated this problem to the NVMe, an ADATA SX8200. With that NVMe installed, system never suspends. Remove the NVMe and it suspends just fine. Moreover, this NVMe locks up the system on boot when running off the latest master branch of firmware. Is it possible to migrate this issue to the firmware-open repo or should I just open a new issue there?

**Comment by jacobgkau** [comment]
> Is it possible to rename and migrate this issue to the firmware-open repo

No, because the firmware-open repo is in the system76 organization and not the pop-os organization.

> I've isolated this problem to the NVMe, an ADATA SX8200.

https://github.com/system76/firmware-open/issues/278 is already open in the firmware-open repo regarding an ADATA SSD. Please follow that issue instead of opening a new one.

**Comment by curiousercreative** [comment]
@jacobgkau apologies, but I jumped to conclusions too quickly. This bug is in fact only reproducible with both an affected ADATA device and kernels newer than 5.13.0-7614-generic and should remain open as an OS or kernel regression bug. I rolled back firmwares and found that suspend continued to not work with the ADATA SX8200 NVMe. I'm now back on the latest galp5 firmware (2021-07-20 which boots) but I boot into this most recent kernel that still suspends. I'd love a more robust solution to continue booting into this kernel, for now I've done the following:
```
$ sudo kernelstub -v -k /boot/vmlinuz-5.13.0-7614-generic -i /boot/initrd.img-5.13.0-7614-generic
```

**Comment by curiousercreative** [comment]
Here's some dmesg output starting before suspend and ending after suspend:
```
[   22.620138] psmouse serio1: Failed to enable mouse on isa0060/serio1
[   25.155788] rfkill: input handler disabled
[   57.766528] wlp0s20f3: deauthenticating from 6c:70:9f:ea:12:47 by local choice (Reason: 3=DEAUTH_LEAVING)
[   59.686251] PM: suspend entry (s2idle)
[   59.691579] Filesystems sync: 0.005 seconds
[   59.958824] Freezing user space processes ... (elapsed 0.002 seconds) done.
[   59.961697] OOM killer disabled.
[   59.961698] Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
[   59.962957] printk: Suspending console(s) (use no_console_suspend to debug)
[   60.218512] psmouse serio1: Failed to disable mouse on isa0060/serio1
[   61.214813] ACPI: EC: interrupt blocked
[   86.578632] ACPI: EC: interrupt unblocked
[   87.096958] mei_me 0000:00:16.0: hbm: dma setup response: failure = 3 REJECTED
[   87.385112] OOM killer enabled.
[   87.385116] Restarting tasks ... 
[   87.391904] mei_hdcp 0000:00:16.0-b638ab7e-94e2-4ea2-a552-d1c54b627f04: bound 0000:00:02.0 (ops i915_hdcp_component_ops [i915])
[   87.396761] done.
[   87.396781] thermal thermal_zone1: failed to read out thermal zone (-61)
[   87.466312] video LNXVIDEO:00: Restoring backlight state
[   87.480301] PM: suspend exit
[   87.528486] Generic FE-GE Realtek PHY r8169-0-2500:00: attached PHY driver (mii_bus:phy_addr=r8169-0-2500:00, irq=MAC)
[   87.732934] r8169 0000:25:00.0 enp37s0: Link is Down
[   93.687672] wlp0s20f3: authenticate with 6c:70:9f:ea:12:47
[   93.691221] wlp0s20f3: send auth to 6c:70:9f:ea:12:47 (try 1/3)
[   93.800209] wlp0s20f3: authenticated
[   93.806402] wlp0s20f3: associate with 6c:70:9f:ea:12:47 (try 1/3)
[   93.807820] wlp0s20f3: RX AssocResp from 6c:70:9f:ea:12:47 (capab=0x1011 status=0 aid=5)
[   93.812202] wlp0s20f3: associated
[   93.878673] IPv6: ADDRCONF(NETDEV_CHANGE): wlp0s20f3: link becomes ready
[   93.902532] wlp0s20f3: Limiting TX power to 17 (17 - 0) dBm as advertised by 6c:70:9f:ea:12:47

**Comment by curiousercreative** [comment]
Here's some dmesg output when suspend succeeds (5.13.0-7614) starting before suspend and ending after suspend:
```
[   30.112335] rfkill: input handler enabled
[   33.390843] rfkill: input handler disabled
[  100.724532] wlp0s20f3: deauthenticating from 6c:70:9f:ea:12:47 by local choice (Reason: 3=DEAUTH_LEAVING)
[  102.635924] PM: suspend entry (s2idle)
[  102.800022] Filesystems sync: 0.164 seconds
[  103.067968] Freezing user space processes ... (elapsed 0.002 seconds) done.
[  103.070630] OOM killer disabled.
[  103.070632] Freezing remaining freezable tasks ... (elapsed 0.000 seconds) done.
[  103.071404] printk: Suspending console(s) (use no_console_suspend to debug)
[  103.331416] psmouse serio1: Failed to disable mouse on isa0060/serio1
[  104.006948] ACPI: EC: interrupt blocked
[  114.739028] ACPI: EC: interrupt unblocked
[  115.394434] mei_me 0000:00:16.0: hbm: dma setup response: failure = 3 REJECTED
[  115.579810] OOM killer enabled.
[  115.579814] Restarting tasks ... 
[  115.583284] mei_hdcp 0000:00:16.0-b638ab7e-94e2-4ea2-a552-d1c54b627f04: bound 0000:00:02.0 (ops i915_hdcp_component_ops [i915])
[  115.586502] done.
[  115.586509] thermal thermal_zone0: failed to read out thermal zone (-61)
[  115.658730] video LNXVIDEO:00: Restoring backlight state
[  115.670756] PM: suspend exit
[  115.730469] Generic FE-GE Realtek PHY r8169-2500:00: attached PHY driver (mii_bus:phy_addr=r8169-2500:00, irq=MAC)
[  115.926687] r8169 0000:25:00.0 enp37s0: Link is Down
[  118.809519] wlp0s20f3: authenticate with 6c:70:9f:ea:12:47
[  118.813204] wlp0s20f3: send auth to 6c:70:9f:ea:12:47 (try 1/3)
[  118.879393] wlp0s20f3: authenticated
[  118.882509] wlp0s20f3: associate with 6c:70:9f:ea:12:47 (try 1/3)
[  118.883865] wlp0s20f3: RX AssocResp from 6c:70:9f:ea:12:47 (capab=0x1011 status=0 aid=5)
[  118.888003] wlp0s20f3: associated
[  118.942973] IPv6: ADDRCONF(NETDEV_CHANGE): wlp0s20f3: link becomes ready
[  118.954435] wlp0s20f3: Limiting TX power to 17 (17 - 0) dBm as advertised by 6c:70:9f:ea:12:47
```

**Comment by curiousercreative** [comment]
Nothing too interesting in the above output, looks identical to me. Looking further back in dmesg output though, this error only appears in the newer kernel that fails to suspend:
```
[    1.461743] i915 0000:00:02.0: [drm] VT-d active for gfx access
[    1.461751] checking generic (90000000 7e9000) vs hw (83000000 1000000)
[    1.461754] checking generic (90000000 7e9000) vs hw (90000000 10000000)
[    1.461756] fb0: switching to i915 from EFI VGA
[    1.461980] i915 0000:00:02.0: vgaarb: deactivate vga console
[    1.493354] input: PNP0C50:00 2808:0101 Mouse as /devices/pci0000:00/0000:00:15.0/i2c_designware.0/i2c-0/i2c-PNP0C50:00/0018:2808:0101.0001/input/input7
[    1.493452] input: PNP0C50:00 2808:0101 Touchpad as /devices/pci0000:00/0000:00:15.0/i2c_designware.0/i2c-0/i2c-PNP0C50:00/0018:2808:0101.0001/input/input8
[    1.493538] hid-generic 0018:2808:0101.0001: input,hidraw0: I2C HID v1.00 Mouse [PNP0C50:00 2808:0101] on i2c-PNP0C50:00
[    1.512914] i915 0000:00:02.0: vgaarb: changed VGA decodes: olddecodes=io+mem,decodes=io+mem:owns=io+mem
[    1.513905] i915 0000:00:02.0: [drm] Finished loading DMC firmware i915/tgl_dmc_ver2_12.bin (v2.12)
[    1.521595] ================================================================================
[    1.521597] UBSAN: shift-out-of-bounds in /build/linux-nx6emy/linux-5.15.11/drivers/gpu/drm/i915/display/intel_opregion.c:388:15
[    1.521601] shift exponent 34 is too large for 32-bit type 'unsigned int'
[    1.521605] CPU: 6 PID: 187 Comm: systemd-udevd Tainted: G           OE     5.15.11-76051511-generic #202112220937~1640185481~21.10~b3a2c21
[    1.521609] Hardware name: System76 Galago Pro/Galago Pro, BIOS 2021-07-20_93c2809 07/20/2021
[    1.521612] Call Trace:
[    1.521614]  <TASK>
[    1.521618]  show_stack+0x52/0x58
[    1.521627]  dump_stack_lvl+0x4a/0x5f
[    1.521634]  dump_stack+0x10/0x12
[    1.521638]  ubsan_epilogue+0x9/0x45
[    1.521642]  __ubsan_handle_shift_out_of_bounds.cold+0x61/0xe9
[    1.521648]  ? icl_ddi_tc_is_clock_enabled+0x4d/0xc0 [i915]
[    1.521869]  intel_opregion_notify_encoder.cold+0x2b/0x45 [i915]
[    1.522076]  intel_sanitize_encoder+0x1e2/0x270 [i915]
[    1.522247]  intel_modeset_setup_hw_state+0x322/0x680 [i915]
[    1.522398]  ? drm_modeset_lock_all_ctx+0x151/0x1c0 [drm]
[    1.522443]  ? drm_warn_on_modeset_not_all_locked.part.0+0x5e/0x90 [drm]
[    1.522478]  intel_modeset_init_nogem+0x2a8/0x510 [i915]
[    1.522625]  ? intel_irq_postinstall+0x38b/0x680 [i915]
[    1.522750]  i915_driver_probe+0x1b7/0x470 [i915]
[    1.522865]  ? mutex_lock+0x13/0x40
[    1.522871]  i915_pci_probe+0x58/0x140 [i915]
[    1.522982]  local_pci_probe+0x48/0x90
[    1.522987]  pci_device_probe+0x115/0x1f0
[    1.522991]  really_probe+0x21b/0x420
[    1.522998]  __driver_probe_device+0x115/0x190
[    1.523002]  driver_probe_device+0x23/0xc0
[    1.523007]  __driver_attach+0xbd/0x1d0
[    1.523009]  ? __device_attach_driver+0x110/0x110
[    1.523012]  bus_for_each_dev+0x7c/0xc0
[    1.523016]  driver_attach+0x1e/0x20
[    1.523020]  bus_add_driver+0x135/0x200
[    1.523025]  driver_register+0x95/0xf0
[    1.523028]  __pci_register_driver+0x68/0x70
[    1.523033]  i915_register_pci_driver+0x23/0x30 [i915]
[    1.523136]  i915_init+0x3b/0xfc [i915]
[    1.523271]  ? 0xffffffffc0aed000
[    1.523273]  do_one_initcall+0x46/0x1d0
[    1.523280]  ? kmem_cache_alloc_trace+0x19e/0x2e0
[    1.523288]  do_init_module+0x62/0x280
[    1.523292]  load_module+0xac9/0xbb0
[    1.523296]  __do_sys_finit_module+0xbf/0x120
[    1.523300]  __x64_sys_finit_module+0x18/0x20
[    1.523302]  do_syscall_64+0x59/0xc0
[    1.523306]  ? do_syscall_64+0x69/0xc0
[    1.523308]  ? syscall_exit_to_user_mode+0x27/0x50
[    1.523312]  ? __x64_sys_read+0x19/0x20
[    1.523316]  ? do_syscall_64+0x69/0xc0
[    1.523319]  ? do_syscall_64+0x69/0xc0
[    1.523321]  ? do_syscall_64+0x69/0xc0
[    1.523323]  ? do_syscall_64+0x69/0xc0
[    1.523325]  ? do_syscall_64+0x69/0xc0
[    1.523327]  ? do_syscall_64+0x69/0xc0
[    1.523330]  entry_SYSCALL_64_after_hwframe+0x44/0xae
[    1.523335] RIP: 0033:0x7f9d6732094d
[    1.523339] Code: 5b 41 5c c3 66 0f 1f 84 00 00 00 00 00 f3 0f 1e fa 48 89 f8 48 89 f7 48 89 d6 48 89 ca 4d 89 c2 4d 89 c8 4c 8b 4c 24 08 0f 05 <48> 3d 01 f0 ff ff 73 01 c3 48 8b 0d b3 64 0f 00 f7 d8 64 89 01 48
[    1.523342] RSP: 002b:00007fffc0561a18 EFLAGS: 00000246 ORIG_RAX: 0000000000000139
[    1.523347] RAX: ffffffffffffffda RBX: 000055d7ae208c30 RCX: 00007f9d6732094d
[    1.523349] RDX: 0000000000000000 RSI: 00007f9d674b23fe RDI: 0000000000000015
[    1.523351] RBP: 0000000000020000 R08: 0000000000000000 R09: 0000000000000000
[    1.523353] R10: 0000000000000015 R11: 0000000000000246 R12: 00007f9d674b23fe
[    1.523355] R13: 000055d7ae206a80 R14: 000055d7ae208bc0 R15: 000055d7ae205e60
[    1.523358]  </TASK>
[    1.523359] ================================================================================
[    1.596389] usb 3-7: new high-speed USB device number 2 using xhci_hcd
[    1.659988] [drm] Initialized i915 1.6.0 20201103 for 0000:00:02.0 on minor 0
[    1.660230] ACPI: video: Video Device [GFX0] (multi-head: yes  rom: no  post: no)
[    1.688595] acpi device:05: registered as cooling_device8
[    1.688686] input: Video Bus as /devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A08:00/LNXVIDEO:00/input/input10
[    1.690594] fbcon: i915drmfb (fb0) is primary device
[    1.690598] fbcon: Deferring console take-over
[    1.690601] i915 0000:00:02.0: [drm] fb0: i915drmfb frame buffer device
```

**Comment by curiousercreative** [comment]
And lastly (for now), my galp5 suspends fine with the ADATA NVMe uninstalled and booted to a live installer USB on the latest kernel.
